Statistics, Exercise 1

Jim O'Regan

Question 1:

In [2]:
# Mean
63 / 300

0.21

Standard error:

$\sqrt{\frac{0.21(1-0.21)}{300}}$

In [6]:
import math
math.sqrt((0.21 * (1 - 0.21)) / 300)

0.023515952032609692

95% confidence limits $=0.21 \pm (1.96 \times 0.02)$

In [8]:
1.96 * 0.02

0.0392

$= 0.21 \pm 0.04$

$=$ **0.17** to **0.25**

Question 2:

In [9]:
533 / 4063

0.13118385429485602

$0.1312 \pm 1.96 \times \sqrt{\frac{0.1312(1-0.312)}{4063}}$

In [10]:
math.sqrt((0.1312 * (1 - 0.1312)) / 4063)

0.005296676057508483

In [11]:
1.96 * 0.0053

0.010388

$0.1312 \pm 0.0103$

**0.1209** to **0.1415**

Exercise 3

In [12]:
x1 = [9.5, 10.5, 9.0, 9.75, 10.0, 13, 10, 13.5, 10, 9.5, 10, 9.75]
x2 = [12.5, 9.5, 13.5, 13.75, 12.0, 13.75, 12.5, 9.5, 12.0, 13.5, 12.0, 12.0]

$H_0: \mu_1 = \mu_2$

$H_1: \mu_1 \ne \mu_2$

In [14]:
import numpy as np
mean1 = np.mean(x1)
mean2 = np.mean(x2)
mean1, mean2

(10.375, 12.208333333333334)

$\bar{X_1} = 10.375$

$\bar{X_2} = 12.2083$

In [26]:
def std(x, mean=None):
    if not mean:
        mean = np.mean(x)
    
    sq_sum = lambda x: (x - mean) ** 2
    n = len(x)
    
    sq_sum = list(map(sq_sum, x))
    return math.sqrt(sum(sq_sum) / (n - 1))


def sem(x):
    s_sq = std(x) ** 2
    n = len(x)
    
    return math.sqrt(s_sq / n)


In [30]:
s1 = std(x1)
s2 = std(x2)
s1, s2

(1.396017061630565, 1.4492683629607002)

In [31]:
len(x1), len(x2)

(12, 12)

$s_1 = 1.396$

$s_2 = 1.449$

$n_1 = 12$

$n_2 = 12$

$X$

$X~~~Y$

Regression model:

$X \rightarrow Y$

Form can be explained as a straight line; if the data is not a straight line, a transformation can be used.

Correlation model:

$X \leftrightarrow Y$

The other model is the correlation model: both $X$ and $Y$ are random.

$X_1, X_2, X_3, \dots X_k, Y$

$Y = \alpha + \beta X + \epsilon$

($Y = \beta_0 + \beta_1 X + \epsilon$)

$\alpha$ : intercept

$\beta$ : coefficient

$\rho$ (random noise?)

We take a sample!

Sample: $y = a + bx$

$\rho$ is estimated by $r$, a 2-dimensional measurement.
In multiple dimensions, it's a correlation between $y$ and $x_1, x_2$, etc.
We need another measurement: $r^2$ (the coefficient of determination; the proportion of explained variance).

Measurement in 2-dimensions is really easy:

$R^2 = (r)^2$

Multiply by 100 for a percentage.

$H_0: \beta = 0$

$H_1: \beta \ne 0$

$> 0$

$< 0$

**BIVARIATE MATERIAL**

$Y = \alpha + \beta X + \epsilon$

$\rho$

Sample:

$\hat{y} = a + bx$

r

$\alpha$: intercept constant

$\beta$: slope regression coefficient

$b = \frac{\Delta y}{\Delta x}$

$r$: correlation coefficient

$-1 \le r \le 1$

$r = 1$: slope upwards

$r = -1$; slope downwards

$r \approx 0$: randomness

$b = \frac{n \Sigma xy - \Sigma x \Sigma y}{n\Sigma x^2 - (\Sigma x)^2}$

$a = $\hat{y} - b \hat{x}$

$= \frac{\Sigma y}{n} - b\frac{\Sigma x}{n}$

$r = \frac{n \Sigma xy - \Sigma x \Sigma y}{\sqrt{[n \Sigma x^2 - (\Sigma x)^2][n\Sigma y^2 - (\Sigma y)^2]}}$


In [15]:
sum_x = 4490
sum_y = 2630
sum_xy = 1237500
sum_xx = 2075550
sum_yy = 760300

top = 12 * sum_xy - (sum_x * sum_y)
bottom = 12 * sum_xx - (sum_x * sum_x)
b = top / bottom

In [16]:
b

0.6407458127040978

In [18]:
a = (sum_y / 12) - (0.64* (sum_x / 12))

In [19]:
a

-20.30000000000004

In [20]:
r_top = (12 * sum_xy) - (sum_x * sum_y)
r_one = (12 * sum_xx - (sum_x * sum_x))
r_two = (12 * sum_yy) - (sum_y * sum_y)

r = r_top / math.sqrt(r_one * r_two)

In [21]:
r

0.9397251970387915

In [22]:
R2 = r * r * 100

In [23]:
R2

88.30834459495955